### PR TITLE
Kernel Selector

### PR DESCRIPTION
**Pull request by blink1073**
REST API-only implementation of the kernelspec, intended to be used by the notebook along with a UI for displaying the kernel selections.

To view the REST API:

Go to http://petstore.swagger.io/#/
Put this in the url box at the top: https://raw.githubusercontent.com/jasongrout/notebook/swagger/notebook/services/api/api.yaml

### CODE REVIEW

File: src/dialog.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+ *    - click : callback to trigger on click
+ *    - class : css classes to add to button.
+ *
+ *
+ *
+ **/
+export
+function modal(options) {
+
+    var modal = $("<div/>")
```
Comment (Carreau) [inline src/dialog.ts]: should jQuery be imported somewhere ?

File: src/dialog.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        }
+        footer.append(button);
+    }
+    dialog_content.append(footer);
+    // hook up on-open event
+    modal.on("shown.bs.modal", function() {
+        setTimeout(function() {
+            footer.find("button").last().focus();
+            if (options.open) {
+                $.proxy(options.open, modal)();
```
Comment (Carreau) [inline src/dialog.ts]: Not that I hate jquery, but I try to decrease it's use. I suppose $.proxy can be changed to `.bind` now.  What do you think ?

Comment (sccolbert) [inline src/dialog.ts]: yes please

Comment (sccolbert) [inline src/dialog.ts]: you dont even need bind or proxy here: `options.open.call(modal)`

Comment (blink1073) [inline src/dialog.ts]: Sorry, this PR isn't ready for review.  I put it up here so I could attach the kernelspec API for my own reference.

Comment (Carreau) [inline src/dialog.ts]: No worries, I was just reading through, and saw the `[wip]`. I'm sure we can do something more integrated with Phosphor than using bootstrap.

**Comment by blink1073** [comment]
Replaced by https://github.com/jupyter/jupyter-js-services/pull/19.

**Pull request closed without merging**